docs/architecture/adr-0001-app-setup-and-goals.md

```diff
@@ -0,0 +1,168 @@
+
+`Model`s also need to know their own requirements to run. In particular, they
+need sufficient information to be able to fill in a `SubmitJob` request for `AWS`
+Batch. This includes resource requirements, a docker container accessible from
+within the hosting AWS account, a compute environment name, and a command to run.
+We require `model`s to carry this information to avoid having to infer it or
+guess at it based on heuristics.
+
+Once we identify that a `model` should run against an image, we'll kick off an
+AWS Batch job with the model's parameters. The `model`'s command _must_ accept a
+STAC item with the `eo` extension as its last parameter.
+
+`Model`s will be renamed to something less overloaded for developer sanity at
+some later date.
+
+### Notifying the API
+
+Since `model`s will be run in some asynchronous workflow, they'll need a way to
+notify the API. The form that notification should take is a [STAC item with the labeling extension](https://github.com/radiantearth/stac-spec/tree/master/extensions/label).
+Reasons to lean on STAC here are similar here to the reasons for leaning on it for new imagery. The API will accept a `POST` to a particular model
+that contains a STAC item with an asset that points to predictions on the new im
+agery.
+
+It's possible that future work could allow discrimination between predictions th
+at need time to run an asynchronous process and models that can do synchronous
+prediction. Since we don't know this is necessary or even possible, we'll assume
+for now that all predictions happen asynchronously and that all updates to the
+Granary API will happen via a later `POST` request.
+
+### Notifying end users
+
+When we create `model`s, we will also create AWS Simple Notification Service
+topics for those models. For our purposes we can think of these as an
+inexhaustible resource, since each AWS account is allowed 100,000 topics. At a
+time in the future when we think we are in danger of going over that limit we
+can think about a new strategy or ask for a service limit increase.
+Whenever a model is updated -- either a new prediction run has been kicked off,
+results are available, or metadata about that model has changed -- we'll send
+an event to the SNS topic for that model. The application will be deployable
+in any AWS account (including a client's account), so we assume that clients
+also have the ability to subscribe to those SNS topics and can subscribe
+to notifications.
+
+### Impact of continuous deployment
+
+There are several consequences of the choice to aim for continuous deployment:
+
+- Continuous integration checks passing on a PR _must_ mean that the branch is
+  safe to deploy. This will require more care around migrations than usual and
+  a more thorough exercise of the application in tests than we have in Raster
+  Foundry. As long as this repository is just an API, that probably means
+  generative testing of workflows modeled through API interaction both for
```

Comment:
I think generative testing will be useful, but also think that the ability to do a high level smoke test of a newly released version of the application will be valuable too. Something that allows us to detect if a critical workflow is compromised due to a change so that we can react before users encounter it.

Reply:
I agree that this is a sensible inclusion, but didn't have too many words to say about it -- see bullet in d01ff86